designer/server/src/test/scala/pl/touk/nussknacker/ui/api/MigrationApiHttpServiceSecuritySpec.scala

```diff
@@ -0,0 +1,123 @@
+    with WithAccessControlCheckingConfigRestAssuredUsersExtensions
+    with WithMockableDeploymentManager
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging {
+
+  "The endpoint for scenario migration between environments when" - {
+    "authenticated should" - {
+      "return response for allowed scenario" in {
+        given()
+          .applicationState(
+            createSavedScenario(exampleScenario, Category1)
+          )
+          .when()
+          .basicAuthAllPermUser()
+          .jsonBody(requestData)
+          .post(s"$nuDesignerHttpAddress/api/migrate")
+          .Then()
+          .statusCode(200)
+          .equalsPlainBody("")
+      }
+    }
+    "not authenticated should" - {
+      "forbid access for user with limited reading permissions" in {
+        given()
+          .applicationState(
+            createSavedScenario(exampleScenario, Category1)
+          )
+          .when()
+          .basicAuthReader()
+          .jsonBody(requestData)
+          .post(s"$nuDesignerHttpAddress/api/migrate")
+          .Then()
+          .statusCode(401)
+          .equalsPlainBody("User doesn't have access to the given category")
+      }
+      "forbid access for user with limited writing permissions" in {
+        given()
+          .applicationState(
+            createSavedScenario(exampleScenario, Category1)
+          )
+          .when()
+          .basicAuthWriter()
+          .jsonBody(requestData)
+          .post(s"$nuDesignerHttpAddress/api/migrate")
+          .Then()
+          .statusCode(401)
+          .equalsPlainBody("The supplied user [writer] is not authorized to access this resource")
+      }
+    }
+    "no credentials were passes should" - {
+      "forbid access" in {
+        given()
+          .applicationState(
+            createSavedScenario(exampleScenario, Category1)
+          )
+          .when()
+          .noAuth()
+          .jsonBody(requestData)
+          .post(s"$nuDesignerHttpAddress/api/migrate")
+          .Then()
+          .statusCode(401)
+          .equalsPlainBody("User doesn't have access to the given category")
+      }
+    }
+  }
+
+  private lazy val sourceEnvironmentId = "DEV"
+
+  private lazy val exampleProcessName = ProcessName("test")
+
+  private lazy val exampleScenario =
+    ScenarioBuilder
+      .withCustomMetaData(exampleProcessName.value, Map("environment" -> "test"))
+      .source("source", "csv-source-lite")
+      .emptySink("sink", "dead-end-lite")
+
+  private lazy val exampleGraph = CanonicalProcessConverter.toScenarioGraph(exampleScenario)
+
+  private def prepareRequestData(scenarioName: String): String =
+    s"""
+       |{
+       |  "sourceEnvironmentId": "$sourceEnvironmentId",
+       |  "processingMode": "Unbounded-Stream",
+       |  "engineSetupName": "Mockable",
+       |  "processName": "${scenarioName}",
+       |  "isFragment": false,
+       |  "processingType": "streaming1",
```

Comment:
We use `TestProcessingType` / `TestCategory` in such cases

Reply:
Fortunately, we dont need anymore `processingType` in request. I removed it from the DTO some time ago but it seems that the excess is ignored during the serialization process.

Reply:
LOL i forgot about that. We should add more restrive requests checks in tests. The part regarding the category is still valid

Reply:
Done